concrete/src/Attribute/Category/SearchIndexer/ExpressSearchIndexer.php

```diff
@@ -20,4 +22,16 @@ public function updateRepository(Entity $previousEntity, Entity $newEntity)
             $this->connection->execute(sprintf('alter table %s rename %s', $previousTable, $newTable));
         }
     }
+
+    protected function isValid(CategoryInterface $category)
+    {
+        $isValid = parent::isValid($category);
+
+        // Express category without a persisted entity are not valid for indexing
+        if ($isValid && $category instanceof ExpressCategory && $category->getExpressEntity()->getHandle() === null) {
```

Comment:
I think this should also check whether `getExpressEntity()` returns an object or any value prior to running `getHandle()` on it.

Reply:
@aembler `getExpressEntity()` always return object because the [constructor](https://github.com/concretecms/concretecms/blob/984b7eb0cf14c8df22b5617ea431f5680ab97d92/concrete/src/Attribute/Category/ExpressCategory.php#L26-L40) requires it.

Reply:
@aembler Any chance to merge this pull request? I got the same error on another website when upgrading from 8 to 9.